lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts

```diff
@@ -117,7 +186,17 @@ class WebMapServiceCatalogItem
     });
   }
 
-  protected async forceLoadMapItems(): Promise<void> {}
+  protected async forceLoadMapItems(): Promise<void> {
+    if (this.invalidLayers.length > 0)
+      throw new TerriaError({
+        sender: this,
+        title: i18next.t("models.webMapServiceCatalogItem.noLayerFoundTitle"),
+        message: i18next.t(
+          "models.webMapServiceCatalogItem.noLayerFoundMessage",
+          { name: getName(this), layers: this.invalidLayers.join(", ") }
+        )
+      });
+  }
```

Comment:
Question: So we fail the dataset even if we have valid layers? Is that the intention?

Reply:
I think for the moment this is fine - as it is what would happen previous to the PR.

But in the future it might be worth showing a warning message for invalid layers, and then just showing valid layers anyway